lib/music_theory/note.rb

```diff
@@ -53,6 +53,19 @@ def distort!(samples)
       end
     end
 
-  end
+    private
 
+    def interpret_frequency_or_pitch_object(frequency_option)
+      if frequency_option.class == Pitch
+        puts "its a Pitch object!!"
+        return frequency_option.frequency
+      elsif frequency_option == Float
+        return frequency_option
+      elsif frequency_option == Integer
```

Comment:
this and line `62` are missing the call to `#class`, might be related to your bug with `nil`